components/SearchField/index.js

```diff
@@ -0,0 +1,141 @@
+import { useState, useEffect } from "react";
+import styled from "styled-components";
+import { AiOutlineSearch, AiOutlineClose } from "react-icons/ai";
+import { keyframes } from "styled-components";
+
+export default function SearchField({ value, handleSearchChange }) {
+  const [showSearch, setShowSearch] = useState(false);
+
+  const handleSearchClick = () => setShowSearch(!showSearch);
+
+  useEffect(() => {
+    const handleResize = () => {
+      setShowSearch(window.innerWidth >= 900);
+    };
+    window.addEventListener("resize", handleResize);
+    handleResize();
+    return () => window.removeEventListener("resize", handleResize);
+  }, []);
+
+  return (
+    <StyledSearchWrapper>
+      {showSearch && (
+        <StyledSearchFieldWrapper className={showSearch ? "show" : ""}>
+          <StyledSearchField
+            type="text"
+            placeholder="Search breeds..."
+            value={value}
+            onChange={handleSearchChange}
+          />
+        </StyledSearchFieldWrapper>
+      )}
+      {showSearch && value && (
+        <StyledClearButton
+          className={showSearch ? "show" : ""}
+          onClick={() => handleSearchChange({ target: { value: "" } })}
```

Comment:
Your handle function currently expects an event object. It's not great to make up a fake event like this. Consider changing your `handleSearchChange` function to only expect the new search term.